src-ui/src/app/components/document-list/filter-editor/filter-editor.component.ts

```diff
@@ -424,9 +424,11 @@ export class FilterEditorComponent implements OnInit, OnDestroy {
     this.subscription = this.textFilterDebounce
       .pipe(debounceTime(400), distinctUntilChanged())
```

Comment:
you can put this into the pipe as well. this will hold the logic together. 
```
.pipe(debounceTime(400), distinctUntilChanged(), filter(query => !query.length || query.length > 2))
```

Reply:
Thanks, agree thats cleaner, I changed it.